presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -124,8 +128,22 @@ private static boolean emptyProjectionIsRequired(Optional<List<ColumnHandle>> pr
                 numberOfRows = result.iterateAll().iterator().next().get(0).getLongValue();
             }
             else {
-                // no filters, so we can take the value from the table info
-                numberOfRows = bigQueryClient.getTable(tableId).getNumRows().longValue();
+                // no filters, so we can take the value from the table info when the object is TABLE
+                TableInfo tableInfo = bigQueryClient.getTable(tableId);
+                if (tableInfo.getDefinition().getType() == TABLE) {
+                    numberOfRows = tableInfo.getNumRows().longValue();
+                }
+                else if (tableInfo.getDefinition().getType() == VIEW) {
```

Comment:
Notice that unless there's some caching on the materialized tables references, then any call to count(*) will trigger a new query and create a new table - incurring additional costs on the users. Another approach is to trigger a sql `SELECT COUNT(*) FROM view` and use the result, which will give BigQuery the option to use some optimizations.

Reply:
As I left to PR description, I considered your "another" approach as well. If BigQuery optimizes the aggregation query internally, I would choose `COUNT(*)` query approach.